### ISSUE TITLE
Error: Returned values aren't valid, did it run Out of Gas?

### ISSUE THREAD

**Issue by 0xEther** [ryanio/truffle-mint-dai#1]
First, thanks for this excellent repo and explanation of using Grid/ganache-cli here: https://medium.com/ethereum-grid/forking-ethereum-mainnet-mint-your-own-dai-d8b62a82b3f7

I'm struggling to mint Dai from the new MCD contract (`0x6b175474e89094c44da98b954eedeac495271d0f`). Using the Truffle tests, I can pass the "should send ether to the Dai contract" test but fail at the next.

The problem piece of code is `await daiContract.methods.balanceOf(account).call();` which throws the error `Error: Returned values aren't valid, did it run Out of Gas?`.

This results in the next assertion failing (`Error: invalid argument 0: hex string without 0x prefix`).

Will look to submit a pull request with a working MCD version if I'm able to get to the bottom of this.

**Comment by ryanio** [comment]
Hey, thanks so much for your initiative to help out with a PR :)

I also see the error `Error: invalid argument 0: hex string without 0x prefix` in the test suite after changing `daiAddress` to `0x6b175474e89094c44da98b954eedeac495271d0f` however if you scroll up the the first error that is encountered before that is `Error: Returned error: VM Exception while processing transaction: revert Dai/not-authorized` as I explained a bit [here](https://medium.com/@ryanghods/hey-kseniya-thanks-for-the-kind-words-797e5f09776d).

Here is my full stack trace:

```
➜  truffle-mint-dai git:(master) truffle test
Using network 'development'.


Compiling your contracts...
===========================
> Compiling ./contracts/ForceSend.sol



  Contract: Truffle Mint Dai
    ✓ should send ether to the Dai contract (6487ms)
    1) should mint Dai for our first 5 generated accounts
    2) "after each" hook: after test for "should mint Dai for our first 5 generated accounts"


  1 passing (7s)
  2 failing

  1) Contract: Truffle Mint Dai
       should mint Dai for our first 5 generated accounts:
     Error: Returned error: VM Exception while processing transaction: revert Dai/not-authorized
      at Object.ErrorResponse (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/web3-core-helpers/src/errors.js:29:1)
      at /Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/web3-core-requestmanager/src/index.js:140:1
      at /Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:112:1
      at XMLHttpRequest.request.onreadystatechange (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/web3-providers-http/src/index.js:96:1)
      at XMLHttpRequestEventTarget.dispatchEvent (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:34:1)
      at XMLHttpRequest._setReadyState (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:208:1)
      at XMLHttpRequest._onHttpResponseEnd (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:318:1)
      at IncomingMessage.<anonymous> (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:289:47)
      at endReadableNT (_stream_readable.js:1145:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)

  2) Contract: Truffle Mint Dai
       "after each" hook: after test for "should mint Dai for our first 5 generated accounts":
     Error: Returned error: Returned error: invalid argument 0: empty hex string
      at Object.ErrorResponse (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/web3-core-helpers/src/errors.js:29:1)
      at /Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/web3-core-requestmanager/src/index.js:140:1
      at /Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:112:1
      at XMLHttpRequest.request.onreadystatechange (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/web3-providers-http/src/index.js:96:1)
      at XMLHttpRequestEventTarget.dispatchEvent (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:34:1)
      at XMLHttpRequest._setReadyState (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:208:1)
      at XMLHttpRequest._onHttpResponseEnd (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:318:1)
      at IncomingMessage.<anonymous> (/Users/rg/.nvm/versions/node/v10.16.2/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:289:47)
      at endReadableNT (_stream_readable.js:1145:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)
```

I couldn't accurately identify an address that could be authorized to mint dai nor find a way to authorize one, so it might be easier to `unlock` an account with an existing dai balance large enough for your needs?

**Comment by 0xEther** [comment]
Thanks for the suggestion. All resolved in #2

**Comment by wswarren12** [comment]
Have tried to unlock the address used in the example with the large DAI amount: 

ganache-cli --fork https://mainnet.infura.io/v3/<infura_id> --gasLimit=0x1fffffffffffff --allowUnlimitedContractSize -e 1000 --networkId=999 --unlock "0x6b175474e89094c44da98b954eedeac495271d0f" --unlock "0x9eb7f2591ed42dee9315b6e2aaf21ba85ea69f8c"

But I keep on getting the following issue when I run Truffle Test: Error: Unknown address - unable to sign transaction for this address: "0x9eb7f2591ed42dee9315b6e2aaf21ba85ea69f8c"

Is this because I'm sending the transaction through an Infura Node?
